backend/src/main/kotlin/hu/bme/sch/cmsch/component/gallery/GalleryService.kt

```diff
@@ -0,0 +1,16 @@
+package hu.bme.sch.cmsch.component.gallery
+
+import jakarta.transaction.Transactional
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.stereotype.Service
+
+@Service
+@ConditionalOnBean(GalleryComponent::class)
+open class GalleryService(
+    private val galleryRepository: GalleryRepository
+) {
+    @Transactional
+    fun savePhoto(galleryEntity: GalleryEntity) = galleryRepository.save(galleryEntity)
+
+    fun fetchAllPhotos(): List<GalleryEntity> = galleryRepository.findAll()
```

Comment:
It must be Transactional to work with postgres db.
I'd add: `@Transactional(readOnly = true)`